Role template: Switching from Config to YAML editor breaks

**Describe the bug**

When switching editor between Config and YAML for the Role Template, it breaks. The issue may be related to https://github.com/rancher/dashboard/issues/12052, although the UI does not break in that case.

**To Reproduce**
- Navigate to `User & Authentication > Role Templates` (`/c/_/auth/roles#GLOBAL`)
- Edit Config for an existing user role, e.g. Administrator
- Switch view to YAML

**Result**

Nothing happens, error in console:

```
entry-helpers.js:40 Error: Cannot find resourceFields for Schema management.cattle.io.templateversion (schemaDefinitions have not been fetched) 
    at get resourceFields (steve-schema.ts:103:1)
```

**Expected Result**
It switches the view, just as it does for all the other resources.

**Screenshots**


https://github.com/user-attachments/assets/46de3314-2d04-47bd-91c0-92817215ee45



**Additional context**

Cluster editor, Workload editor and probably all the other work as expected.

--- a/shell/components/ResourceDetail/index.vue
+++ b/shell/components/ResourceDetail/index.vue
@@ -332,7 +332,11 @@ export default {
 
       const queryDiff = Object.keys(diff(neu, old));
 
-      if ( queryDiff.includes(MODE) || queryDiff.includes(AS)) {
+      if (Object.keys(neu).length <= 0) {
+        return;
+      }
+
+      if (queryDiff.includes(MODE) || queryDiff.includes(AS)) {
         this.$fetch();
       }
     },

